src/CMakeLists.txt

```diff
@@ -67,57 +67,54 @@ foreach(SUB_DIR ${TOP_LEVEL_SUBDIRS})
 endforeach()
 
 add_library(
-  ${CHECKPOINT_LIBRARY}
+  ${MAGISTRATE_LIBRARY}
   ${HEADER_FILES} ${SOURCE_FILES}
 )
 
-add_library(${CHECKPOINT_LIBRARY_NS} ALIAS ${CHECKPOINT_LIBRARY})
+add_library(${MAGISTRATE_LIBRARY_NS} ALIAS ${MAGISTRATE_LIBRARY})
 
-target_compile_features(${CHECKPOINT_LIBRARY} PUBLIC cxx_std_17)
+target_compile_features(${MAGISTRATE_LIBRARY} PUBLIC cxx_std_17)
 
 include(CMakePrintHelpers)
 
 if (${kokkos_DIR_FOUND})
   if (KokkosKernels_DIR_FOUND)
-    message(STATUS "Checkpoint: Kokkos kernels enabled")
+    message(STATUS "Magistrate: Kokkos kernels enabled")
     set(KERNELS 1)
     # Kokkos Kernels historically installed its EXPORT targets in the Kokkos:: namespace, and several
     # downstream packages import it as such, so it's not readily changed.
     # Meanwhile, Trilinos installs it under the more consistent KokkosKernels:: namespace.
     # Account for both possible routes of acquiring Kokkos Kernels here
     if(TARGET KokkosKernels::kokkoskernels)
-      target_link_libraries(${CHECKPOINT_LIBRARY} PUBLIC KokkosKernels::kokkoskernels)
+      target_link_libraries(${MAGISTRATE_LIBRARY} PUBLIC KokkosKernels::kokkoskernels)
     else()
-      target_link_libraries(${CHECKPOINT_LIBRARY} PUBLIC Kokkos::kokkoskernels)
+      target_link_libraries(${MAGISTRATE_LIBRARY} PUBLIC Kokkos::kokkoskernels)
     endif()
   else()
-    message(STATUS "Checkpoint: Kokkos kernels disabled")
+    message(STATUS "Magistrate: Kokkos kernels disabled")
     set(KERNELS 0)
   endif()
   target_compile_definitions(
-    ${CHECKPOINT_LIBRARY} PUBLIC
+    ${MAGISTRATE_LIBRARY} PUBLIC
     KOKKOS_ENABLED_CHECKPOINT=1 KOKKOS_KERNELS_ENABLED=${KERNELS}
   )
-  message(STATUS "Checkpoint: Kokkos enabled")
-  target_link_libraries(${CHECKPOINT_LIBRARY} PUBLIC Kokkos::kokkos)
+  message(STATUS "Magistrate: Kokkos enabled")
+  target_link_libraries(${MAGISTRATE_LIBRARY} PUBLIC Kokkos::kokkos)
 endif()
 
-target_link_libraries(${CHECKPOINT_LIBRARY} PUBLIC ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${MAGISTRATE_LIBRARY} PUBLIC ${CMAKE_THREAD_LIBS_INIT})
 
 if(CODE_COVERAGE AND CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-  message(STATUS "Checkpoint: enabling code coverage generation")
+  message(STATUS "Magistrate: enabling code coverage generation")
   # Add required flags (GCC & LLVM/Clang)
   target_compile_options(
-    ${CHECKPOINT_LIBRARY} PUBLIC
+    ${MAGISTRATE_LIBRARY} PUBLIC
     -O0        # no optimization
     -g         # generate debug info
     --coverage # sets all required flags
   )
-  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.13)
```

Comment:
Right now we require CMake 3.17 (vt requires 3.23, we should also bump the version for magistrate)

Reply:
https://github.com/DARMA-tasking/magistrate/issues/354 created